### PR TITLE
[SC] Send transactions using unconfirmed outputs

### LINKED ISSUE #2801: [SC] Block validation fails in SmartContractFormatRule with multiple transactions each referencing the one before

### PR DESCRIPTION
**Pull request by codingupastorm**
Closes #2801 and #2800

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/UnspentOutputSet.cs [hunk 1]
```diff
@@ -76,7 +76,7 @@ public void TrySetCoins(UnspentOutputs[] coins)
             }
         }
 
-        public IList<UnspentOutputs> GetCoins(ICoinView utxo)
```
Comment (rowandh) [inline src/Stratis.Bitcoin.Features.Consensus/UnspentOutputSet.cs]: 😮

Comment (codingupastorm) [reply src/Stratis.Bitcoin.Features.Consensus/UnspentOutputSet.cs]: Good point

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Consensus/Rules/SmartContractFormatRule.cs [hunk 2]
```diff
@@ -43,9 +43,9 @@ public override Task RunAsync(RuleContext context)
                 if (!transaction.IsSmartContractExecTransaction())
                     return Task.CompletedTask;
 
-                Money transactionFee = transaction.GetFee(((UtxoRuleContext)context).UnspentOutputSet);
+                // Money transactionFee = transaction.GetFee(((UtxoRuleContext)context).UnspentOutputSet);
```
Comment (rowandh) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Consensus/Rules/SmartContractFormatRule.cs]: 😎

**Comment by rowandh** [comment]
I know CoinViewRule is notoriously hard to test... but is there a way we could create an integration test to demonstrate #2801?

**Comment by codingupastorm** [comment]
Good call @rowandh did so and realised the controllers weren't allowing it properly. Should be all good now, see newest commit for changes

**Pull request merged**